File: pipeline/problems/a125600d-fed9-4358-93bc-66a308345d68/Outputs/CodeContentFiles/NodeJS/solution.js
class Solution {
    static locatePairPositions(values, required) {
        const seen = new Map();
        for (let i = 0; i < values.length; i++) {
            const num = values[i];
            const complement = required - num;
            if (seen.has(complement)) {
                return [seen.get(complement), i];
            }
            seen.set(num, i);
        }
        return [];
    }
}

File: pipeline/problems/a125600d-fed9-4358-93bc-66a308345d68/Outputs/CodeContentFiles/Cpp/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class solution{
public:
    vector<int> locatePairPositions(vector<int>& values,int required){
        unordered_map<int,int> seen;
        for(int i=0;i<(int)values.size();i++){
            int num=values[i];
            int complement=required-num;
            if(seen.find(complement)!=seen.end()){
                return {seen[complement],i};
            }
            seen[num]=i;
        }
        return {};
    }
};